src/instructlab/model/download.py

```diff
@@ -48,38 +283,28 @@
 @click.pass_context
 @clickext.display_params
 def download(ctx, repository, release, filename, model_dir, hf_token):
-    """Download the model(s) to train"""
-    click.echo(f"Downloading model from {repository}@{release} to {model_dir}...")
-    if hf_token == "" and "instructlab" not in repository:
-        raise ValueError(
-            """HF_TOKEN var needs to be set in your environment to download HF Model.
-            Alternatively, the token can be passed with --hf-token flag.
-            The HF Token is used to authenticate your identity to the Hugging Face Hub."""
+    downloader = None
+
+    if is_oci_repo(repository):
+        downloader = OCIDownloader(
+            repository=repository, release=release, download_dest=model_dir
         )
+
+    else:
+        downloader = HFDownloader(
+            repository=repository,
+            release=release,
+            download_dest=model_dir,
+            filename=filename,
+            hf_token=hf_token,
+            ctx=ctx,
+        )
+
```

Comment:
What if the repo is not OCI and not HF? We need to handle that case, or is it handled somewhere I can't find it? We need to validate `repository` further. So you need if, elif, else here if not done.

```
@click.option(
    "--repository",
    default=DEFAULTS.MERLINITE_GGUF_REPO,  # TODO: add to config.yaml
    show_default=True,
    help="Hugging Face repository of the model to download.",
)
```

Also, the help section must be updated.

Reply:
added the check for repository type and updated the help msg